packages/psss/src/utils/PersistGateProvider.js

```diff
@@ -8,23 +8,25 @@ import { PersistGate } from 'redux-persist/integration/react';
 import PropTypes from 'prop-types';
 
 function initPersistor(store) {
-  const persistor = persistStore(store, null);
+  const persistor = persistStore(store);
 
-  // if you run into problems with redux state, call "purge()" in the dev console
-  if (window.localStorage.getItem('queuePurge')) {
-    persistor.purge();
-    window.localStorage.setItem('queuePurge', '');
-  }
+  // Handle rememberMe flow.
+  // The PSSS:activeSession is used to determine if it is a new or existing active session
+  if (!window.sessionStorage.getItem('PSSS:activeSession')) {
+    const rememberMe = window.localStorage.getItem('PSSS:rememberMe');
 
-  window.purge = () => {
-    window.localStorage.setItem('queuePurge', 'true');
-    window.location.reload();
-  };
+    // If it is a new session and the user has not ticked rememberMe, then remove the user data from localStorage
+    if (rememberMe === 'false') {
+      persistor.purge();
```

Comment:
Ideally we'd like to keep their email address, but throw away everything else (so a partial rather than full purge). That could be easier through a redux persist rehydration transform, rather than through session and local storage (and would also mean the action handling all went through redux rather than setting two different things). This is a nice to have, so up to you whether you give it a go

Reply:
I initially set it up using a transform (in the same way as AdminPanel) but there is a big advantage to doing it this way which is:
- the redux data gets stored for the duration of the session even if rememberMe is not ticked. This is great because the user can still refresh the page and not get logged out or loose their page

On the admin panel if you did not tick the rememberMe checkbox and refresh you get booted out which could be frustrating

Reply:
Ah yep cool. For my own learning: what are the boundaries of a session?